src/window.rs

```diff
@@ -767,6 +767,19 @@ impl Window {
     pub fn request_user_attention(&self, request_type: Option<UserAttentionType>) {
         self.window.request_user_attention(request_type)
     }
+
+    /// Returns the visible state of the window. This is `true` if some part of the window
+    /// is visible on screen and false otherwise. This is useful for pausing rendering
+    /// when the window isn't visible since some platforms may behave strangely if you
+    /// try to draw to an occluded window.
```

Comment:
Can you give an example of this? E.g. why is it not enough to just check the `visible` property (via. [`-isVisible`](https://developer.apple.com/documentation/appkit/nswindow/1419132-visible?language=objc))

Reply:
@madsmtm `visible` returns true even when the window in question is fully occluded by other windows, i.e it's not hidden in the window manager. `AnimationCoreKit` on Mac as of some recent Big Sur versions on amd64 (can't confirm this) and the M1 chip stops periodical resource reclamation tasks as a battery saving measure when the window is hidden *or* fully occluded. As described in the linked `wgpu` issue this means that graphics applications may exhaust resources like memory and `MTLDrawable`'s when the window becomes fully occluded, causing an error within Metal.